File: umbrella-insurance-react-ui/src/models/exchangeRates/ExchangeRate.ts
import { Unit } from "../units/v1/Unit";

export class ExchangeRate {
    id?: number;
    unit1?: Unit;
    unit2?: Unit;
    unit1ToUnit2Ratio?: number;
    constructor(obj?: any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.unit1) {
                this.unit1 = new Unit(obj.unit1);
            }
            if(obj?.unit2) {
                this.unit2 = new Unit(obj.unit2);
            }
            if(obj?.unit1ToUnit2Ratio) {
                this.unit1ToUnit2Ratio = obj.unit1ToUnit2Ratio;
            }

        }
    }
}

File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/ExchangeRate.java
package com.umbrella.insurance.core.models.entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "exchange_rates", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "exchange_rates_unique", columnNames = {"unit_id_1", "unit_id_2"})
})
public class ExchangeRate {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "exchange_rate_id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "unit_id_1")
    private Unit unit1;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "unit_id_2")
    private Unit unit2;

    @NotNull
    @Column(name = "unit_1_to_unit_2_ratio", nullable = false)
    private Double unit1ToUnit2Ratio;

}